app/cruds/cruds_raffle.py

```diff
@@ -0,0 +1,429 @@
+                        models_raffle.TypeTicket.raffle
+                    ),
+                    joinedload(models_raffle.Tickets.user),
+                    joinedload(models_raffle.Tickets.lot),
+                )
+            )
+        )
+        .scalars()
+        .all()
+    )
+    filtered_results = [
+        result for result in results if result.type_ticket.raffle_id == raffle_id
+    ]
+    return filtered_results
+
+
+async def get_ticket_by_id(
+    ticket_id: str,
+    db: AsyncSession,
+) -> models_raffle.Tickets | None:
+    result = await db.execute(
+        select(models_raffle.Tickets).where(models_raffle.Tickets.id == ticket_id)
+    )
+    return result.scalars().first()
+
+
+async def get_ticket_by_userid(
+    user_id: str,
+    db: AsyncSession,
+) -> list[models_raffle.Tickets] | None:
+    result = await db.execute(
+        select(models_raffle.Tickets)
+        .where(models_raffle.Tickets.user_id == user_id)
+        .options(
+            joinedload(models_raffle.Tickets.type_ticket).joinedload(
+                models_raffle.TypeTicket.raffle
+            ),
+            joinedload(models_raffle.Tickets.user),
+            joinedload(models_raffle.Tickets.lot),
+        )
+    )
+    return result.scalars().all()
+
+
+async def delete_ticket(
+    db: AsyncSession,
+    ticket_id: str,
+):
+    """Delete a ticket from database by id"""
+
+    await db.execute(
+        delete(models_raffle.Tickets).where(models_raffle.Tickets.id == ticket_id)
+    )
+    await db.commit()
+
+
+async def get_users_cash(db: AsyncSession) -> list[models_raffle.Cash]:
+    result = await db.execute(
+        select(models_raffle.Cash).options(selectinload(models_raffle.Cash.user))
+    )
+    return result.scalars().all()
+
+
+async def get_cash_by_id(db: AsyncSession, user_id: str) -> models_raffle.Cash | None:
+    result = await db.execute(
+        select(models_raffle.Cash)
+        .where(models_raffle.Cash.user_id == user_id)
+        .options(selectinload(models_raffle.Cash.user))
+    )
+    return result.scalars().first()
+
+
+async def create_cash_of_user(
+    db: AsyncSession, cash: models_raffle.Cash
+) -> models_raffle.Cash:
+    db.add(cash)
+    try:
+        await db.commit()
+        return cash
+    except IntegrityError as err:
+        await db.rollback()
+        raise err
+
+
+async def add_cash(db: AsyncSession, user_id: str, amount: float):
+    result = await db.execute(
+        select(models_raffle.Cash).where(models_raffle.Cash.user_id == user_id)
+    )
+    balance = result.scalars().first()
+    if balance is not None:
+        await db.execute(
+            update(models_raffle.Cash)
+            .where(models_raffle.Cash.user_id == user_id)
+            .values(user_id=balance.user_id, balance=balance.balance + amount)
+        )
+        try:
+            await db.commit()
+        except IntegrityError:
+            await db.rollback()
+            raise ValueError("Error during cash edition")
```

Comment:
Returning the error will help during debugging 
```suggestion
        except IntegrityError as err:
            await db.rollback()
            raise err
```